crates/contracts/.gitignore

```diff
@@ -0,0 +1 @@
+src/alloy
```

Comment:
nit: small comment explaining that this directory only contains auto generated code would be nice.